rest/route/annotations.go

```diff
@@ -213,3 +216,104 @@ func (h *annotationByTaskHandler) Run(ctx context.Context) gimlet.Responder {
 
 	return gimlet.NewJSONResponse(res)
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// PUT /rest/v2/task/{task_id}/annotation
+
+type annotationByTaskPutHandler struct {
+	taskId    string
+	execution int
+
+	user       gimlet.User
+	annotation *annotations.TaskAnnotation
+	sc         data.Connector
+}
+
+func makePutAnnotationsByTask(sc data.Connector) gimlet.RouteHandler {
+	return &annotationByTaskPutHandler{
+		sc: sc,
+	}
+}
+
+func (h *annotationByTaskPutHandler) Factory() gimlet.RouteHandler {
+	return &annotationByTaskPutHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *annotationByTaskPutHandler) Parse(ctx context.Context, r *http.Request) error {
+	var err error
+
+	h.taskId = gimlet.GetVars(r)["task_id"]
+	if h.taskId == "" {
+		return gimlet.ErrorResponse{
+			Message:    "task ID cannot be empty",
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+	execution := r.URL.Query().Get("execution")
+	if execution != "" {
+		h.execution, err = strconv.Atoi(execution)
+		if err != nil {
+			return gimlet.ErrorResponse{
+				Message:    fmt.Sprintf("Invalid execution: '%s'", err.Error()),
+				StatusCode: http.StatusBadRequest,
+			}
+		}
+	} else {
+		h.execution = -1
```

Comment:
You can't update all annotations, so  this doesn't work. If h.execution == "", you want to update whatever the latest execution is (you can move getting the task above this, so if execution isn't specified you can set to t.Execution)

Reply:
I added a check to UpdateAnnotation()